doc/realPSD-advancedtut.Rmd

```diff
@@ -364,13 +364,26 @@ If you still remember the OU example in the [quick tutorial](http://htmlpreview.
   /// @return An `ou::UFun` object.
   template<class Type>
   UFun<Type> make_Ufun(int N, objective_function<Type>* obj) {
-    return UFun<Type>(N);
+    // pick method
+    DATA_STRING(method);
```

Comment:
Hi @mlysy, I have updated the `devel` branch to include the latest changes I made based on your suggestion above. But there are two things to note:

1. I guess `DATA_MATRIX(extra_arg)` should be optional but not required, otherwise, our built-in models (OU, SHOF, SHOW) will encounter errors about reading data `extra_arg` when calling `TMB::MakeADFun()` since we don't supply any additional data at all. So my approach is to add an overloaded class constructor `UFun<Type>::UFun(cRefMatrix<Type>& f, cRefMatrix<Type>& extra_arg)` and hide the line `DATA_MATRIX(extra_arg);` inside `make_Ufun` where the returned object is `UFun<Type>(f)` instead of `UFun<Type>(f, extra_arg)`. Please see the updated advanced tutorial on this part.

2. Do you think `extra_arg` is a good name? It means extra argument(s) which may sound like parameters instead of data. Would it be better if we call this variable `extra_data`? What do you think?

Thanks.

Reply:
`extra_arg` is just a generic name for an "extra argument" if you have one.  Otherwise, you skip this entirely.  In reality, if your extra arguments are called e.g., `p` and `q`, then you would have e.g., `UFun<Type>::UFun(cRefMatrix<Type>& f, cRefMatrix<Type>& p, cRefMatrix<Type>& q)`, etc.